advanced/wallets/react-wallet-v2/src/views/SessionProposalModal.tsx

```diff
@@ -226,17 +226,22 @@ export default function SessionProposalModal() {
     }
   }, [])
 
-  const namespaces = buildApprovedNamespaces({
-    proposal: proposal.params,
-    supportedNamespaces
-  })
+  const namespaces = useMemo(() => {
+    try {
+      // the builder throws an exception if required namespaces are not supported
+      return buildApprovedNamespaces({
+        proposal: proposal.params,
+        supportedNamespaces
+      })
+    } catch (e) {}
```

Comment:
does this need a toast?

Reply:
if namespaces are not built there is message in the proposal
![image](https://github.com/WalletConnect/web-examples/assets/43912948/77e30b15-97dc-4075-a525-ece80382bbf8)